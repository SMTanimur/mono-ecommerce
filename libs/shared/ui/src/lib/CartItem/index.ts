import { CartItem, CartItemProps } from './CartItem';

export { CartItem };
export type { CartItemProps };
