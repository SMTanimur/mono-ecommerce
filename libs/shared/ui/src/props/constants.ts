import { ButtonProps } from '@ecommerce/shared/ui/Button';
import {CartItemProps} from "@ecommerce/shared/ui/CartItem";
import {ProductSidebarProps} from "@ecommerce/shared/ui/ProductSidebar";
import {CartSidebarProps} from "@ecommerce/shared/ui/CartSidebar";
import {CategoryCardProps} from "@ecommerce/shared/ui/CategoryCard";
import {IconProps} from "@ecommerce/shared/ui/Icon";
import {ItemCardProps} from "@ecommerce/shared/ui/ItemCard";
import {NavButtonProps} from "@ecommerce/shared/ui/NavButton";
import {ProductGalleryProps} from "@ecommerce/shared/ui/ProductGallery";

import cat from './cat.jpg';
import cat2 from './cat2.jpeg';

console.log(cat2)

export const PRODUCT: ProductSidebarProps['product'] = {
  slug: 'some-name',
  price: 65,
  name: 'Some name',
  images: {
    data: [
      {
        attributes: {
          url: cat,
        },
      },
      {
        attributes: {
          url: cat2,
        },
      },
    ],
  },
  smallDescription: 'Small Description',
  description: 'Description',
  information: [
    {
      title: 'Title1',
      content: 'content1',
    },
  ],
};

export const ITEM: ItemCardProps['item'] = {
  slug: 'cat-cat',
  price: 65,
  name: 'cat cat',
  images: {
    data: [
      {
        attributes: {
          url: cat,
        },
      },
      {
        attributes: {
          url: cat2,
        },
      },
    ],
  },
  smallDescription: 'smallDescription',
  description: 'description',
};

export const BUTTON_DEFAULT_PROPS: ButtonProps = {
  children: 'hello',
  tag: 'button',
  size: 'base',
  type: 'primary',
  round: 'full',
};

export const CART_ITEM_DEFAULT_PROPS: CartItemProps = {
  inCartAmount: 1,
  item: PRODUCT,
  handleCartMinusClick: () => {
    console.log('cart minus');
  },
  handleCartPlusClick: () => {
    console.log('cart plus');
  },
  handleRemoveFromCart: () => {
    console.log('cart remove');
  },
  toggleFavorite: () => {
    console.log('favorite');
  },
};

export const CART_SIDEBAR_DEFAULT_PROPS: CartSidebarProps = {
  cartTotal: 100,
  isUser: true,
  handleCheckoutClick: async () => {
    console.log('checkout');
  },
  isLoading: false,
};

export const CATEGORY_CARD_DEFAULT_PROPS: CategoryCardProps = {
  category: {
    smallDescription:
      'smalll des',
    image: {
      data: {
        attributes: {
          url: cat,
        },
      },
    },
    slug: '1',
  },
};
export const ICON_DEFAULT_PROPS: IconProps = {
  icon: 'cart',
};

export const ITEM_CARD_DEFAULT_PROPS: ItemCardProps = {
  item: ITEM,
  size: 'base',
  handleCartClick: () => console.log('cart clicked'),
  handleAmountCartClick: () => console.log('amount cart clicked'),
  handleFavoriteClick: () => console.log('favorite clicked'),
};
export const NAV_BUTTON_DEFAULT_PROPS: NavButtonProps = {
  type: 'heart',
};
export const PRODUCT_GALLERY_DEFAULT_PROPS: ProductGalleryProps = {
  images: [
    {
      attributes: {
        url: cat,
      },
    },
    {
      attributes: {
        url: cat2,
      },
    },
  ],
};
