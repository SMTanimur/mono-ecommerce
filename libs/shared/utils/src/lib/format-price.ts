export const formatPrice = (price: number) =>
  `${Math.max(price, 0).toFixed(2)}$`;
