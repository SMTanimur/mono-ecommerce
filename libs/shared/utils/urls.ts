export const getApiUrl = (path = ''): string =>
  `${process.env['NX_STRAPI_IMG_URL'] ?? ''}${path}`;

export const getImageUrl = (path = ''): string =>
  `${process.env['NX_STRAPI_URL'] ?? 'http://localhost:4200'}${path}`;

export const createItemHref = (id: string): string => 
    `/i/${id}`;
