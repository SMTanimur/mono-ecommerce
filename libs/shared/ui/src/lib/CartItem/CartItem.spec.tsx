import '@testing-library/jest-dom';
import { RenderResult, render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';

import { formatPrice } from '@ecommerce/shared/utils/format-price';

import { CART_ITEM_DEFAULT_PROPS } from '../../props/constants';
import { CartItem } from './CartItem';

describe('CartItem', () => {
  let renderResult: RenderResult;

  beforeEach(() => {
    renderResult = render(<CartItem {...CART_ITEM_DEFAULT_PROPS} />);
  });

  it('should render successfully', () => {
    const { baseElement } = renderResult;

    const tree = create(<CartItem {...CART_ITEM_DEFAULT_PROPS} />).toJSON();
    expect(tree).toMatchSnapshot();

    expect(baseElement).toBeTruthy();
  });

  it('main elements are exists and properly formatted', () => {
    const { item, inCartAmount } = CART_ITEM_DEFAULT_PROPS;
    const { price, name, smallDescription } = item;

    const properPrice = formatPrice(price * (inCartAmount || 0));

    screen.getByText(name);
    screen.getByText(properPrice);
    screen.getByText(smallDescription);
  });
});
