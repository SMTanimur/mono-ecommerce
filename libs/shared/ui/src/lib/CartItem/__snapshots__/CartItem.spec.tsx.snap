// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem should render successfully 1`] = `
<div
  className="w-full flex"
>
  <a
    href="/i/some-name"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <div
      className="relative aspect-square w-28 object-cover bg-gray-100"
    >
      Next image stub
    </div>
  </a>
  <div
    className="ml-8 w-full flex flex-col"
  >
    <div
      className="w-full flex flex-wrap items-center justify-between gap-x-1"
    >
      <a
        className="text-base font-semibold uppercase hover:(underline)"
        href="/i/some-name"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Some name
      </a>
      <span
        className="font-bold"
      >
        65.00$
      </span>
    </div>
    <small
      className="mt-1"
    >
      Small Description
    </small>
    <div
      className="mt-auto flex gap-8 items-center flex-wrap md:(gap-4) lg:(gap-8)"
    >
      <div
        className="h-8 flex"
      >
        <button
          className="h-full w-8 relative group flex items-center justify-center border rounded-l-xl transition hover:(bg-black text-white border-black)"
          onClick={[Function]}
          title="remove 1 item"
          type="button"
        >
          <div
            aria-hidden="true"
            className="h-1/2 w-1/2 relative group-hover:(filter invert brightness-0)"
          >
            <svg
              className="feather feather-minus "
              fill="none"
              height={15}
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width={15}
            >
              <g
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<line x1=\\"5\\" y1=\\"12\\" x2=\\"19\\" y2=\\"12\\"></line>",
                  }
                }
              />
            </svg>
          </div>
        </button>
        <span
          className="w-8 flex items-center justify-center border-t border-b"
        >
          1
        </span>
        <button
          className="h-full w-8 relative group flex items-center justify-center border rounded-r-xl transition hover:(bg-black text-white border-black)"
          onClick={[Function]}
          title="add 1 item"
          type="button"
        >
          <div
            aria-hidden="true"
            className="h-1/2 w-1/2 relative group-hover:(filter invert brightness-0)"
          >
            <svg
              className="feather feather-plus "
              fill="none"
              height={15}
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width={15}
            >
              <g
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<line x1=\\"12\\" y1=\\"5\\" x2=\\"12\\" y2=\\"19\\"></line><line x1=\\"5\\" y1=\\"12\\" x2=\\"19\\" y2=\\"12\\"></line>",
                  }
                }
              />
            </svg>
          </div>
        </button>
      </div>
      <div
        className="gap-8 items-center hidden text-xs xs:(flex) sm:(text-base) md:(text-xs gap-4) lg:(text-base gap-8)"
      >
        <button
          className="hover:(underline)"
          onClick={[Function]}
          type="button"
        >
          Remove item
        </button>
        <button
          className="hover:(underline)"
          onClick={[Function]}
          type="button"
        >
          Add to favorites
        </button>
      </div>
    </div>
  </div>
</div>
`;
